### PR TITLE
Add git signoff in apply_patches helper function

### PR DESCRIPTION
**Pull request by slevenick**
Add git signoff in apply_patches helper function. InSpec requires this, should be no negative effect

-----------------------------------------------------------------
# [all]
## [inspec]
Test

### CODE REVIEW

File: .ci/magic-modules/helpers.sh [hunk 1]
```diff
@@ -16,5 +16,5 @@ function apply_patches {
   # We keep all the changes (--soft).
   git reset --soft "$(git merge-base HEAD "$4")"
   # Then we commit again.
-  git commit -m "$2" --author="$3" || true  # don't crash if no changes
+  git commit -m "$2" --author="$3" --signoff || true  # don't crash if no changes
```
Comment (ndmckinley) [inline .ci/magic-modules/helpers.sh]: Question - I think this is going to say "signed off by Modular Magician" - is that what you want?  Or do you prefer signed-off-by the actual author?

Comment (slevenick) [reply .ci/magic-modules/helpers.sh]: I'm not sure it matters. But wouldn't the original author have to perform the commit to have it signed off by them?

Comment (ndmckinley) [reply .ci/magic-modules/helpers.sh]: Well, git has to believe that the original author did it - but you see here where --author="$3"; we can probably pretend to be that person in this script, if it does matter.

Comment (ndmckinley) [inline .ci/magic-modules/helpers.sh]: Ideally you should make sure that the signoff line doesn't appear in downstream PR messages, as well.

Comment (ndmckinley) [reply .ci/magic-modules/helpers.sh]: e.g. https://github.com/modular-magician/inspec-gcp/pull/54#issue-242386371

**Pull request closed without merging**